### PR TITLE
Extended dataset cheatsheet in documentation: `MNISTSuperpixels`, `FAUST`, `ShapeNet`, `ModelNet`, and `CoMA`

### PR DESCRIPTION
**Pull request by EdisonLeeeee**

### CODE REVIEW

File: torch_geometric/datasets/modelnet.py [hunk 1]
```diff
@@ -50,6 +50,22 @@ class ModelNet(InMemoryDataset):
...
+            :header-rows: 1
+
+            * - #graphs
+              - #nodes
+              - #edges
+              - #features
+              - #classes
+            * - 4,011
+              - 9,882.5
+              - 38,831.8
```
Comment (rusty1s) [inline torch_geometric/datasets/modelnet.py:66]:
```suggestion
              - ~9,882.5
              - ~38,831.8
```

File: torch_geometric/datasets/shapenet.py [hunk 2]
```diff
@@ -54,6 +54,22 @@ class ShapeNet(InMemoryDataset):
...
+            :widths: 10 10 10 10 10
+            :header-rows: 1
+
+            * - #graphs
+              - #nodes
+              - #edges
+              - #features
+              - #classes
+            * - 16,881
+              - 2,616.2
```
Comment (rusty1s) [inline torch_geometric/datasets/shapenet.py:69]:
```suggestion
              - ~2,616.2
```

**Review by rusty1s — APPROVED** [review]
Amazing, thank you!

File: torch_geometric/datasets/modelnet.py [hunk 3]
```diff
@@ -50,6 +50,22 @@ class ModelNet(InMemoryDataset):
             :obj:`torch_geometric.data.Data` object and returns a boolean
             value, indicating whether the data object should be included in the
             final dataset. (default: :obj:`None`)
+
+    Stats:
+        .. list-table::
+            :widths: 10 10 10 10 10
+            :header-rows: 1
+
+            * - #graphs
```
Comment (EdisonLeeeee) [inline torch_geometric/datasets/modelnet.py:59]: There should be two datasets, modelnet10 and modelnet40. Pls don't merge now.

**Pull request merged**